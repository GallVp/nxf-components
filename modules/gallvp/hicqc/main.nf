process HICQC {
    tag "$meta.id"
    label 'process_single'

    // Bioconda recipe is pending: https://github.com/bioconda/bioconda-recipes/pull/55921
    container "docker.io/gallvp/hic_qc:v1.3"

    input:
    tuple val(meta), path(bam)

    output:
    tuple val(meta), path("*.pdf")  , emit: pdf
    tuple val(meta), path("*.html") , emit: html
    path "versions.yml"             , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def prefix = task.ext.prefix ?: "$meta.id"
    def args = task.ext.args ?: ''
    """
    mkdir -p matplotlib/config
    mkdir -p matplotlib/cache

    export MPLCONFIGDIR=./matplotlib/config
    export XDG_CACHE_HOME=./matplotlib/cache

    hic_qc.py \\
        $args \\
        -b $bam \\
        --outfile_prefix "$prefix"

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        hic_qc.py: \$(hic_qc.py --version)
    END_VERSIONS
    """

    stub:
    """
    touch "${meta.id}.pdf"
    touch "${meta.id}.html"

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        hic_qc.py: \$(hic_qc.py --version)
    END_VERSIONS
    """
}
