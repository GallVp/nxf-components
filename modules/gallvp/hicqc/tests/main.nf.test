nextflow_process {

    name "Test Process HICQC"
    script "../main.nf"
    process "HICQC"

    tag "modules"
    tag "modules_gallvp"
    tag "hicqc"

    test("hongyang - bam") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file('https://github.com/Plant-Food-Research-Open/assemblyqc/raw/4f9fba392cf7212d61c6245fd72339e09164ce44/tests/hic/testdata/HYh1h2_Chr01_6000000_6015000.bam', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.versions,
                    file(process.out.pdf[0][1]).name,
                    file(process.out.html[0][1]).name,
                    file(process.out.html[0][1]).readLines()[13..16]
                    ).match()
                }
            )
        }

    }

    test("hongyang - bam - stub") {

        options '-stub'

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file('https://github.com/Plant-Food-Research-Open/assemblyqc/raw/4f9fba392cf7212d61c6245fd72339e09164ce44/tests/hic/testdata/HYh1h2_Chr01_6000000_6015000.bam', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
