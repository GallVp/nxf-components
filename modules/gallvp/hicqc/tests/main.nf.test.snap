{
    "hongyang - bam": {
        "content": [
            [
                "versions.yml:md5,d8f5bd4c7db89f17c564f3b1c8770d7f"
            ],
            "test_qc_report.pdf",
            "test_qc_report.html",
            [
                "<td style=\"text-align: left;\">BAM File</td>",
                "<td style=\"text-align: left;\">HYh1h2_Chr01_6000000_6015000.bam</td>",
                "</tr>",
                "<tr>"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.6"
        },
        "timestamp": "2025-05-27T15:07:01.177127"
    },
    "hongyang - bam - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.pdf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.html:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,d8f5bd4c7db89f17c564f3b1c8770d7f"
                ],
                "html": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.html:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "pdf": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.pdf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d8f5bd4c7db89f17c564f3b1c8770d7f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.6"
        },
        "timestamp": "2025-05-27T11:12:42.637405"
    }
}